### PR TITLE
[v0.1] Basic Count Statistics - Item Counter implemented

### PR DESCRIPTION
**Pull request by lucasferrari07**
Function itemCounts returns a Map with the count of each element in the array received as parameter. For every element in the array, if it already exists in the Map, the code store it's current value to be incrementedin the Map. Otherwise, it is stored with a initial value of 1

### CODE REVIEW

File: textalyze.js [hunk 1]
```diff
@@ -14,8 +14,10 @@
 function itemCounts(array) {
   let counts = new Map();
 
-  // Your code here.
-  // Run 'npm test' to see what tests need to pass.
+  for (let a of array) {
```
Comment (lucasferrari07) [inline textalyze.js]: I now see that `a` should be a const as it's value will not be changed.

File: textalyze.js [hunk 2]
```diff
@@ -14,10 +6,17 @@
 function itemCounts(array) {
   let counts = new Map();
 
-  // Your code here.
-  // Run 'npm test' to see what tests need to pass.
+  for (const a of array) {
```
Comment (jfarmer) [inline textalyze.js]: Exercise: how would you write this as a single call to [reduce](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce)?

File: textalyze.js [hunk 3]
```diff
@@ -14,10 +6,17 @@
...
+    counts.has(a) ? counts.set(a, counts.get(a) + 1) : counts.set(a, 1);
+  }
 
   return counts;
 }
 
+const elements = ["a", "a", "a", "b", "b", "c"];
```
Comment (jfarmer) [inline textalyze.js]: We'd like this file to be both included as a module and run via `npm start`.  Check out https://stackoverflow.com/questions/4981891/node-js-equivalent-of-pythons-if-name-main for how to do this.

You want

```javascript
if (require.main == module) {
  const array = ['a', 'a', 'a', ...];
}
```

File: textalyze.js [hunk 4]
```diff
@@ -14,10 +6,17 @@
...
+console.log(`Count for [${elements}] is...`);
+itemCounts(elements).forEach((count, elem) => {
```
Comment (jfarmer) [inline textalyze.js]: Try to be consistent — you use `forEach` here but `for ... in ...` above.  The AirBnb style guide will tell you to use `forEach`.

File: textalyze.js (hunk 3 above)
Comment (jfarmer) [reply textalyze.js]: Make sure you're using ESlint.  I've set up the project to automatically enforce a style guide using AirBnb's recommended style.

You can set up your editor to do it automatically or you can manually run `npx eslint` to run the version included with the project on the command line.

File: textalyze.js [hunk 5]
```diff
@@ -14,10 +6,17 @@
...
+  console.log(elem, count);
+});
```
Comment (jfarmer) [inline textalyze.js]: Check out [console.table](https://developer.mozilla.org/en-US/docs/Web/API/Console/table) — it's useful for quickly printing out tables in debugging situations.

**Review by jfarmer — APPROVED** [review]
Try to write good commit messages and good PR messages. A good commit message means that someone else could read the commit message and understand what is happening.

Read:
- https://chris.beams.io/posts/git-commit/
- https://github.com/erlang/otp/wiki/writing-good-commit-messages
- https://medium.com/@andrewhowdencom/anatomy-of-a-good-commit-message-acd9c4490437

Rather than saying "pull request for first iteration" or "implemented itemCounts" without anything else, treat it like an actual project.  Commit messages and PR messages are a kind of documentation.

You want to say something like:

```
Implement itemCounts

itemCounts takes in an Array and returns a Map containing the count
of each item in the Array.  For example, itemCounts(['a', 'a', 'b']) would
return the Map { 'a': 2, 'b': 1 }.
```

The sooner you get into this habit the sooner it will stop feeling like homework.

Later, when the projects become more substantial, we will _need_ good commit messages and PR descriptions in order to give you a quick, thorough code review.

**Pull request merged**